### ISSUE TITLE
Freeing resources while quitting issue

### ISSUE THREAD

**Issue by simast** [tesselode/kira#12]
Seeing this error in console on every quit (and it takes indeed 1 second for program to exit):

> Kira failed to free up resources after 1000 milliseconds, giving up

Seems this is the bare minimum example that reproduces this issue for me:

```rust
let mut manager = AudioManager::new(AudioManagerSettings::default())?;
let music_track = manager.add_sub_track(SubTrackSettings::default())?;
```

Tested on macOS.

**Edit:** Realized that the issue seems to be related to the fact I keep the sub-track reference in the same object as audio manager:

```rust
pub struct Audio {
    manager: AudioManager,
    music_track: SubTrackHandle,
}
```

Not sure how I can workaround this? I do need a reference to this track handle for later use.

**Comment by tesselode** [comment]
I'm able to reproduce this issue. I'll have to think about a good solution to this. I'll get back to you when I have one.

**Comment by tesselode** [comment]
So!

I know why this is happening.

Bad news: this is hard to fix with Kira's current architecture
Good news: I'm in the middle of rearchitecting kira in a way that will fix this problem!

In the meantime, I don't have a great recommendation other than: drop the `SubTrackHandle` before dropping the `AudioManager`. I know it's a pain, so I will be working to make resource management more ergonomic.

**Comment by simast** [comment]
👍 not a problem if the fix is on the way as part of the refactor you are doing. Awesome library and API btw.

**Comment by tesselode** [comment]
thanks!

**Comment by nathande** [comment]
@simast Have you tried putting the fields in the other order?:
```
pub struct Audio {
    music_track: SubTrackHandle,
    manager: AudioManager,
}
```
The objects are dropped in the order specified in the struct definition if I remember correctly, which is something I ended up having to look into when interfacing with a Rust binding to an audio library written in C.

**Comment by tesselode** [comment]
Fixed in 31642b0.
